readthedocs/core/utils/__init__.py

```diff
@@ -186,8 +186,8 @@ def prepare_build(
                 project_slug=project.slug,
                 version_slug=version.slug,
             )
-            options['countdown'] = 5 * 60
```

Comment:
How does this code actually cancel the build? it seems it's just setting options on the task, and an `error`, but still triggering the build? 

Also seems like these aren't required, since it should take this from the task?

Reply:
This PR does not change the flow of this chunk of code. I just move these numbers into Django settings.

That said, this code is run in the webs when the build is triggered and it immediately sets a 5 minutes delay because we already know this project is concurrency limited at the moment. If we delete this code, builders will grab each of these triggered tasks and retry them immediately. So, we are just short-circuiting the process here.

Note that it _does not cancel_ the build, it just adds a countdown to be delayed 5 minutes. The error message is to communicate this delay to the user.

Reply:
Ah, interesting. That makes the comment above make a lot more sense. 👍